File: eMovieFinder/eMovieFinder.Model/Entities/Actor.cs
using eMovieFinder.Model.Utilities;
using System;

namespace eMovieFinder.Model.Entities
{
    public class Actor : TimeStampObjectViewModel
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public byte[]? Image { get; set; }
        public DateTime BirthDate { get; set; }
        public int? CountryId { get; set; }
        public string IMDbLink { get; set; }
        public string Biography { get; set; }
        public string? FormattedActorBirthDate => BirthDate.ToString("yyyy-MM-dd");
        public string? FormattedActorBirthPlace => Country?.CountryName;
        public string? FormattedActorRealName => $"{FirstName} {LastName}";
        public virtual Country Country { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_mobile/lib/src/models/entities/actor.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:emovie_finder_mobile/src/models/utilities/time_stamp_object.dart';
import 'country.dart';

part 'actor.g.dart';

@JsonSerializable()
class Actor extends TimeStampObject {
  final int? id;
  final String? firstName;
  final String? lastName;
  final dynamic image;
  final DateTime? birthDate;
  final int? countryId;
  final String? imDbLink;
  final String? biography;
  final String? formattedActorBirthDate;
  final String? formattedActorBirthPlace;
  final String? formattedActorRealName;
  final Country? country;

  Actor({
    super.creationDate,
    super.modifiedDate,
    super.createdById,
    super.modifiedById,
    super.formattedCreationDate,
    super.formattedModifiedDate,
    this.id,
    this.firstName,
    this.lastName,
    this.image,
    this.birthDate,
    this.countryId,
    this.imDbLink,
    this.biography,
    this.formattedActorBirthDate,
    this.formattedActorBirthPlace,
    this.formattedActorRealName,
    this.country
  });

  factory Actor.fromJson(Map<String, dynamic> json) => _$ActorFromJson(json);
  Map<String, dynamic> toJson() => _$ActorToJson(this);
}